### ISSUE TITLE
Upgrade framework to Material Design 2

### ISSUE THREAD

**Issue by johnleider** [vuetifyjs/vuetify#6676]
# 2.0 Tasklist
Below is an exhaustive list of the remaining features and functionality that will be implemented before entering beta in Q2.

#### How do I run the Alpha?
Vuetify has moved to an install pattern similar to vue-router and vuex. Below is the baseline procedure for bootstrapping your application.
```js
// v1.5
// src/plugins/vuetify.js
import Vue from 'vue'
import Vuetify from 'vuetify'

Vue.use(Vuetify, { ... })
```

```js
//  v2.0
// src/plugins/vuetify.js
import Vue from 'vue'
import Vuetify from 'vuetify'

Vue.use(Vuetify)

export default new Vuetify({ ... })
```
```js
// v2.0
// src/main.js
import Vue from 'vue'
import vuetify from './plugins/vuetify'

new Vue({
  vuetify
}).$mount('#app')
```

#### Is there an upgrade guide?
Not until beta. For individual breaking changes/upgrades required, please step through the Alpha releases listed below.

#### What does a completed item mean?
A completed item is one that has:
- Converted styles from stylus to sass
- Converted tests from avoriaz to vue-test-utils
- Updated according to MD2 Spec (if applicable)
- Convert components / tests to Typescript (if applicable)

#### What happens to components not in MD2 spec?
Version 2 will be a culmination of MD1 and MD2 specs. Components will use their most updated specification version. These have been marked on the corresponding documentation pages with a link to their corresponding specifications.
![image](https://user-images.githubusercontent.com/9064066/53848065-97633f00-3f78-11e9-82cc-e6491ba2cfde.png)

#### When will v2 enter **beta**?
Our current release target is Q2 2019. Beta will take place once all components have been converted to our new structure and updated to MD2 (if applicable). From now until release, we will be providing at minimum weekly updates with our progression.

#### Should I use this in production?
No

#### Should I report bugs that I find in the **alpha**?
No. We will open up bug reporting once we move to **beta**

#### Where can I check out the alpha documentation?
https://next.vuetifyjs.com

# Releases
- **[Alpha.20](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.20)** - 5/14
- **[Alpha.19](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.19)** - 5/7
- **[Alpha.18](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.18)** - 4/30
- **[Alpha.17](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.17)** - 4/23
- **[Alpha.16](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.16)** - 4/17
- **[Alpha.15](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.15)** - 4/17
- **[Alpha.14](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.14)** - 4/16
- **[Alpha.13](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.13)** - 4/9
- **[Alpha.12](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.12)** - 4/2
- **[Alpha.11](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.11)** - 3/29
- **[Alpha.10](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.10)** - 3/19
- **[Alpha.9](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.9)** - 3/13
- **[Alpha.8](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.8)** - 3/12
- **[Alpha.7](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.7)** - 3/5
- **[Alpha.6](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.6)** - 2/26
- **[Alpha.5](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.5)** - 2/19
- **[Alpha.4](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.4)** - 2/13
- **[Alpha.3](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.3)** - 1/29
- **[Alpha.2](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.2)** - 1/24
- **[Alpha.1](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.1)** - 1/22
- **[Alpha.0](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0-alpha.0)** - 1/15

# Roadmap
### Components
- [ ] ~~VGrid~~ - #7126~ **pushed to v2.1~

### New Components
- [ ] ~~VDiscovery~~ **on hold** - #6731
- [ ] ~~VDualList~~ **on hold**
- [ ] ~~VMore~~ **on hold**
- [ ] ~~VTestimonial~~ **on hold**
- [ ] ~~VUpload~~ **on hold**

# Completed
#### Alpha.20
- [x] VOverflowBtn - 170ede3
- [x] VTextField - #7157
#### Alpha.19
- [x] VAutocomplete - #7136
- [x] VCheckbox - #7136 
- [x] VCombobox - #7136
- [x] VInput - #7136
- [x] VProgressCircular - #7110
- [x] VRadioGroup - #7136
- [x] VSelect - #7136
- [x] VTextarea - #7136
#### Alpha.18
- [x] VMenu - #7098
- [x] VParallax - #7042
- [x] VProgressLinear - #7057
- [x] VSnackbar - #7013
#### Alpha.17
- [x] VStepper - #7041
#### Alpha.14
- [x] VDatePicker - 59bbe2a
- [x] VNavigationDrawer - #6947
#### Alpha.13
- [x] VCarousel - #6924
- [x] VDialog - a6820bb
- [x] VPagination - a8b7fc1
- [x] VSpeedDial - #6938
#### Alpha.12
- [x] VForm - d9b744e
- [x] VSwitch - #6851
- [x] VTimePicker - #6869
- [x] VTreeview - #6849
#### Alpha.11
- [x] VApp - #6718
- [x] VBottomSheet - #6786
- [x] VBreadcrumbs - #6778 
- [x] VCard - #6788
- [x] VImg - a312292
- [x] VSparkline - #6827
#### Alpha.10
- [x] bootable - #6735
- [x] VBottomNav - #6741
- [x] VExpansionPanel - #6720
- [x] VList - #6648
#### Alpha.8
- [x] VCounter - f1245f7
- [x] VDataIterator
- [x] VDataTable
- [x] VFooter - #6711
- [x] VHover - 30fdd3e
- [x] VResponsive - e501b808b
- [x] VSubheader - #6713
- [x] VSystemBar - #6701
- [x] VToolbar - #6701
- [x] VWindow - a5039f2
#### Alpha.7
- [x] VItemGroup - #6631
- [x] VRangeSlider
- [x] VSlider - #6424 
- [x] VTabs - #6606 
- [x] VTimeline - #6656
- [x] Typography - #6653
#### Alpha.6
- [x] VBadge - #6514 
- [x] VBanner - #6520 
- [x] VDivider - #6582 
- [x] VLabel - #6584 
- [x] VMessages - #6585 
- [x] VRating - bc91c41
- [x] VSheet - bddb379
#### Alpha.5 - Services (core) #6464
- [x] Application
- [x] Breakpoint
- [x] GoTo
- [x] Icons
- [x] Lang
- [x] Theme
#### Alpha.4
- [x] VAvatar - #6433 
- [x] VChip - #6303 
- [x] VIcon - #6439
#### Alpha.0
- [x] VAlert - #6107 
- [x] VBtn - #6037 
- [x] VBtnToggle - #6037
- [x] VTooltip - #6111

#### Beta.8
- [x] VFileInput
#### Alpha.19
- [x] VColorPicker - #6988
#### Alpha.8
- [x] VAppBar - #6701
#### Alpha.6
- [x] VBanner - #6520
#### Alpha.4
- [x] VChipGroup - #6303 
#### Alpha.3
- [x] VSlideGroup - #6240 

#### Alpha.10 26c5596
- [x] click-outside
- [x] resize
- [x] ripple
- [x] scroll
- [x] touch
